tests/entities/test_trace_info.py

```diff
@@ -42,3 +48,22 @@ def test_to_proto():
     assert isinstance(tag_2, ProtoTraceTag)
     assert tag_2.key == "k" * 250
     assert tag_2.value == "v" * 250
+
+
+def test_to_dict(trace_info):
+    trace_as_dict = trace_info.to_dict()
+    assert trace_as_dict == {
+        "request_id": "request_id",
+        "experiment_id": "test_experiment",
+        "timestamp_ms": 0,
+        "execution_time_ms": 1,
+        "status": "OK",
+        "request_metadata": {
+            "foo": "bar",
+            "k" * 1000: "v" * 1000,
+        },
+        "tags": {
+            "baz": "qux",
+            "k" * 2000: "v" * 2000,
+        },
+    }
```

Comment:
Awesome that we're adding this. I think this still passes on `master` because the `status` field of `TraceInfo` is being set to a `TraceStatus` object in https://github.com/mlflow/mlflow/pull/12044/files#diff-5aee524692b8b8837b083f08c50a7feb4ebc2d192b97548527ad6978173bc99dR16. 

If we also add https://github.com/mlflow/mlflow/pull/12044/files#r1605960450, then we will prevent recurrence.

Reply:
doh!